### ISSUE TITLE
block-scoped-var fails for nested destructuring

### ISSUE THREAD

**Issue by divmain** [eslint/eslint#3019]
The issue is demonstrated by the following two examples.

**1:**
```javascript
/* eslint-disable no-console */
function f (input) {
  const [{prop}, second] = input;
  console.log(prop, second);  // <--
}
f([{ prop: "a" }, "b"]);
```
**2:**
```javascript
/* eslint-disable no-console */
function g (input) {
  const {first: [el], second} = input; // <--
  console.log(el, second); // <--
}
g({ first: ["a"], second: "b" });
```

These two examples compile and execute correctly using Babel.  However, ESlint will report a `block-scoped-var` warning on line 4 of the first example, and for both locations where `el` shows up in the second example.

Looking [here](https://github.com/eslint/eslint/blob/master/lib/rules/block-scoped-var.js#L26-L39), there is no case for where `parent.type === "ArrayPattern"` or `parent.type === "ObjectPattern"`.  I would have submitted a PR, however this check needs to be context sensitive: the immediate parent is either an `ArrayPattern` or `ObjectPattern` **and** the parent's parent is one or more levels of `ArrayPattern`/`ObjectPattern` (checking parent's parent's parent, and so on), ultimately terminating in `VariableDeclarator` or `FunctionExpression`.

That sort of check does not appear to be supported by `estraverse`.  So, considering the largish nature of a potential solution, I didn't want to make the attempt without first soliciting feedback.

Thanks!

**Edit:** May be related to #2253.  If identical, please feel free to close.  However, the discussion there led me to believe a separate problem was being addressed.

**Comment by nzakas** [comment]
Thanks, can you please provide the actual ESLint output you're seeing for each example?

**Comment by gyandeeps** [comment]
@divmain Are you still seeing this error or the issue can be closed?

**Comment by gyandeeps** [comment]
Closing because of no response.
Feel free to reopen if its still a issue.

**Comment by divmain** [comment]
Sorry for the radio silence, I've been traveling for the last week an a half.  I upgraded both ESlint and `babel-eslint` and this issue went away.  Thanks!

**Comment by IanVS** [comment]
Thanks for reporting back your success.
